week3/homework/src/index.js

```diff
@@ -1,3 +1,192 @@
 'use strict';
 
 // TODO: Write the homework code in this file
+
+const express = require('express');
+
+const uuid = require('uuid/v4');
+
+const {
+  readFile: _readFile,
+  writeFile: _writeFile
+} = require('fs');
+
+const {
+  promisify
+} = require('util');
+
+const readFile = promisify(_readFile);
+const writeFile = promisify(_writeFile);
+
+const app = express();
+
+//USE BUILT-IN JSON middleware to automatically parse JSON
+app.use(express.json());
+
+const TODO_FILE = 'todo.json';
+
+function readTodos() {
+  return readFile(TODO_FILE, 'utf-8').then(JSON.parse, () => []);
+}
+
+function writeTodos(todos) {
+  return writeFile(TODO_FILE, JSON.stringify(todos));
+}
+
+
+//CREATE TODO
+app.post('/todos', async (req, res) => {
+  const newTodo = req.body;
+  newTodo.id = uuid();
+  newTodo.done = false;
+  const todos = await readTodos();
+  todos.push(newTodo);
+  await writeTodos(todos);
+  res.json(todos);
+  res.end();
+
+});
+
+//READ ALL TODOs
+app.get('/todos', async (req, res) => {
+  const todos = await readTodos();
+  res.json(todos);
+  res.end();
+});
+
+//READ TODO BY ID
+app.get('/todos/:id', async (req, res) => {
+  try {
+    const id = req.params.id;
+    const todos = await readTodos();
+    const todo = todos.find(x => x.id === id);
+    if (typeof todo === 'undefined') {
+      res.json({
+        'Error Message': 'NO such ID !!'
+      });
+    }
+    res.json(todo);
```

Comment:
Since you already returned a response above if the `todo` was not found, you should not call `res.json` again. This part of the code should be unreachable if we don't need/want it to be executed. What you could do, is `return` on line 64. Does that make sense?

Reply:
can i use res.send on line 64 when ID is not exist !!?